crates/mpt/src/db/mod.rs

```diff
@@ -307,11 +352,27 @@ where
         Ok(int_slot)
     }
 
-    fn block_hash(&mut self, _: U256) -> Result<B256, Self::Error> {
-        // match self.db.block_hashes.entry(number) {
-        //     Entry::Occupied(entry) => Ok(*entry.get()),
-        //     Entry::Vacant(_) => anyhow::bail!("Block hash for number not found"),
-        // }
-        unimplemented!("Block hash not implemented; Need to unroll the starting block hash for this operation.")
+    fn block_hash(&mut self, block_number: U256) -> Result<B256, Self::Error> {
+        // The block number is guaranteed to be within the range of a u64.
+        let u64_block_number: u64 = block_number.to();
+
+        // Fetch the block header from the preimage fetcher.
+        let mut block_hash = self.parent_block_hash;
+        let mut header = (self.header_by_hash_fetcher)(block_hash)?;
```

Comment:
Do we need these parenthesis?

```suggestion
        let mut header = self.header_by_hash_fetcher(block_hash)?;
```

Reply:
Yep, we do. Rust won't compile w/o it.